src/components/InitialNickname/InitialNickname.tsx

```diff
@@ -0,0 +1,29 @@
+'use client';
+
+import { STORAGE_KEY } from '@/constants/storage';
+import { useLocalStorage } from '@/hooks/useLocalStorage';
+import { generateRandomNickname } from '@/utils/randomNickname';
+import React from 'react';
+
+interface InitialNicknameProps {
+  children: React.ReactNode;
+}
+
+const InitialNickname = ({ children }: InitialNicknameProps) => {
+  const { setItem, getItem } = useLocalStorage();
+
+  const initialNickname = () => {
+    const nickname = getItem(STORAGE_KEY.NICKNAME);
+
+    if (nickname === '') {
+      const newNickname = generateRandomNickname();
+      setItem(STORAGE_KEY.NICKNAME, newNickname);
+    }
+  };
+
+  initialNickname();
+
+  return <>{children}</>;
+};
```

Comment:
Hydration 오류를 확인해봤는데 localStorage는 클라이언트에서만 사용할 수 있기 때문에 서버에서 렌더링될 때 localStorage를 참조할 수 없어서 Hydration Error가 발생한다고하네요.

useEffect와 useState을 이용해서 hydration 과정 이후에 children이 렌더링 되도록 바꿔주면 해결됩니다!

```suggestion
const InitialNickname = ({ children }: InitialNicknameProps) => {
  const { setItem, getItem } = useLocalStorage();
  const [isInitialized, setInitialized] = useState(false);

  useEffect(() => {
    const nickname = getItem(STORAGE_KEY.NICKNAME);

    if (nickname === '') {
      const newNickname = generateRandomNickname();
      setItem(STORAGE_KEY.NICKNAME, newNickname);
    }

    setInitialized(true);
  }, [getItem, setItem]);

  if (!isInitialized) {
    return null;
  }

  return <>{children}</>;
};
```